system/common.xml

```diff
@@ -75,21 +75,37 @@
 						<echo message="This branch does not exist!"/>
 					</then>
 				</if>
+				<if>
```

Comment:
Instead of if, should this be `else block`? Also, I would like to have `git checkout` logic once.
pseudocode:
```
run git rev-parse ${stf_branch} // failonerror="false" and get stf_sha1
If code=128 {
    git rev-parse origin/${stf_branch} // set failonerror="true" and set stf_sha2
     if code2 =128 {
           // error. fail the build. if set failonerror="true" works as expected, we do not need this if block
      } else {
          // success. Set stf_sha
         <property stf_sha value="${stf_sha2}"/>
      }
}
if  stf_sha is not set {
      // set it to stf_sha1
      <property stf_sha value="${stf_sha1}"/>
}
run git checkout -q -f ${stf_sha}
```

Reply:
I changed the logic according to the pseudocode, it is indeed a cleaner approach, and I also rerun on the grinder(#1514,#1515,#1516).

Reply:
Thanks a lot, @xius666 !